ratis-shell/src/main/java/org/apache/ratis/shell/cli/RaftUtils.java

```diff
@@ -64,9 +64,13 @@ public static RaftPeerId getPeerId(String host, int port) {
    * @return return a raft client
    */
   public static RaftClient createClient(RaftGroup raftGroup) {
-    RaftProperties properties = new RaftProperties();
-    RaftClientConfigKeys.Rpc.setRequestTimeout(properties,
+    RaftProperties defaults = new RaftProperties();
+    RaftClientConfigKeys.Rpc.setRequestTimeout(defaults,
         TimeDuration.valueOf(15, TimeUnit.SECONDS));
+
+    // Since ratis-shell support GENERIC_COMMAND_OPTIONS, here we should
+    // merge these options to raft properties to make it work.
+    RaftProperties systems = new RaftProperties(System.getProperties());
```

Comment:
@AngersZhuuuu , let's simply use `forEach` instead of changing `RaftProperties`.
```java
@@ -67,6 +68,12 @@ public final class RaftUtils {
     RaftProperties properties = new RaftProperties();
     RaftClientConfigKeys.Rpc.setRequestTimeout(properties,
         TimeDuration.valueOf(15, TimeUnit.SECONDS));
+
+    // Since ratis-shell support GENERIC_COMMAND_OPTIONS, here we should
+    // merge these options to raft properties to make it work.
+    final Properties sys = System.getProperties();
+    sys.stringPropertyNames().forEach(key -> properties.set(key, sys.getProperty(key)));
+
     ExponentialBackoffRetry retryPolicy = ExponentialBackoffRetry.newBuilder()
```

Reply:
DOne